src/main/java/com/example/springbootbookshop/service/impl/UserServiceImpl.java

```diff
@@ -35,7 +38,15 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
         User user = userMapper.toUserModel(requestDto);
         user.setPassword(passwordEncoder.encode(requestDto.password()));
         user.setRoles(getUserRole());
-        return userMapper.toUserDto(userRepository.save(user));
+        User savedUser = userRepository.save(user);
+        createAndSetNewCartToUser(savedUser);
+        return userMapper.toUserDto(savedUser);
+    }
+
+    private void createAndSetNewCartToUser(User user) {
+        Cart cart = new Cart();
+        cart.setUser(user);
+        cartRepository.save(cart);
```

Comment:
Don't need to save cart here

Reply:
i check it, if i don`t save cart when i register new user new cart for this user doesn`t created